### PR TITLE
hwmv2: soc: nxp: port kinetis family to hardware model v2

### PR DESCRIPTION
**Pull request by danieldegrasse**
Ports Kinetis SOC family and boards to HWMv2. *Note- this PR depends on #68777 for the definition of `SOC_PART_NUMBER`*

### CODE REVIEW

File: soc/nxp/kinetis/Kconfig.soc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2016 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_KINETIS
+	bool
+
+config SOC_FAMILY
+	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/Kconfig.soc:8]:
```suggestion
```

File: soc/nxp/kinetis/Kconfig.defconfig [hunk 2]
```diff
@@ -8,4 +10,6 @@ config ADC_INIT_PRIORITY
 	default 80 if ADC_MCUX_ADC16_ENABLE_EDMA
 	depends on ADC
 
-source "soc/soc_legacy/arm/nxp_kinetis/*/Kconfig.defconfig.series"
+rsource "*/Kconfig.defconfig"
+
+endif # SOC_FAMILY_KINETIS
```
Comment (nordicjm) [inline soc/nxp/kinetis/Kconfig.defconfig:15]: Seems empty lines are missing from these files

File: soc/nxp/kinetis/k2x/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Copyright 2024 NXP
+
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if SOC_SERIES_KINETIS_K2X
+
+config SOC_SERIES
+	default "k2x"
```
Comment (nordicjm) [inline soc/nxp/kinetis/k2x/Kconfig.defconfig:14]: this goes in the Kconfig.soc file

File: soc/nxp/kinetis/kl2x/Kconfig.defconfig [hunk 4]
```diff
@@ -1,13 +1,14 @@
 # Kinetis KL2x series configuration options
 
-# Copyright (c) 2017, NXP
+# Copyright (c) 2017, 2024 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 if SOC_SERIES_KINETIS_KL2X
 
 config SOC_SERIES
 	default "kl2x"
```
Comment (nordicjm) [inline soc/nxp/kinetis/kl2x/Kconfig.defconfig:9]: Kconfig.soc

File: soc/nxp/kinetis/kwx/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Kinetis KWx series configuration options
+
+# Copyright (c) 2017, 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_KINETIS_KWX
+
+config SOC_SERIES
+	default "kwx"
```
Comment (nordicjm) [inline soc/nxp/kinetis/kwx/Kconfig.defconfig:9]: Kconfig.soc

File: boards/nxp/frdm_k22f/Kconfig.frdm_k22f [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2018, Prevas A/S
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_FRDM_K22F
+	select SOC_MK22F51212
+	
+config SOC_PART_NUMBER
+	default "MK22FN512VLH12"
```
Comment (nordicjm) [inline boards/nxp/frdm_k22f/Kconfig.frdm_k22f:8]: This should be set in Kconfig.soc

File: boards/nxp/frdm_k64f/Kconfig.frdm_k64f [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_FRDM_K64F
+	select SOC_MK64F12
+
+config SOC_PART_NUMBER
+	default "MK64FN1M0VLL12"
```
Comment (nordicjm) [inline boards/nxp/frdm_k64f/Kconfig.frdm_k64f:7]: As above

File: boards/nxp/frdm_k82f/Kconfig.frdm_k82f [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019 SEAL AG
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_FRDM_K82F
+	select SOC_MK82F25615
+
+config SOC_PART_NUMBER
+	default "MK82FN256VLL15"
```
Comment (nordicjm) [inline boards/nxp/frdm_k82f/Kconfig.frdm_k82f:8]: As above

File: boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2017, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_FRDM_KL25Z
+	select SOC_MKL25Z4
+
+config SOC_PART_NUMBER
+	default "MKL25Z128VLK4"
```
Comment (nordicjm) [inline boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:8]: As above

File: boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2017, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_FRDM_KW41Z
+	select SOC_MKW41Z4
+
+config SOC_PART_NUMBER
+	default "MKW41Z512VHT4"
```
Comment (nordicjm) [inline boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z:8]: As above

File: boards/nxp/hexiwear/Kconfig.hexiwear [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+	bool
+	select SOC_MK64F12
+
+config BOARD_HEXIWEAR_MKW40Z4
+	bool
+	select SOC_MKW40Z4
+
+config SOC_PART_NUMBER
+	default "MK64FN1M0VDC12" if BOARD_HEXIWEAR_MK64F12
+	default "MKW40Z160VHT4" if BOARD_HEXIWEAR_MKW40Z4
```
Comment (nordicjm) [inline boards/nxp/hexiwear/Kconfig.hexiwear:15]: As above

File: boards/nxp/hexiwear/doc/index.rst [hunk 12]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+| RTT       | on-chip    | console                             |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | soc flash                           |
++-----------+------------+-------------------------------------+
+| TRNG      | on-chip    | entropy                             |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/hexiwear_kw40z/hexiwear_kw40z_defconfig``
```
Comment (nordicjm) [inline boards/nxp/hexiwear/doc/index.rst:385]: Path needs updating

File: boards/nxp/hexiwear/doc/index.rst [hunk 13]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
+   :goals: flash
+
+The Segger RTT console is only available during a debug session. Use ``attach``
+to start one:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
```
Comment (nordicjm) [inline boards/nxp/hexiwear/doc/index.rst:476]: Board names need updating

File: boards/nxp/hexiwear/doc/index.rst [hunk 14]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+   :goals: attach
+
+Run telnet as shown earlier, and you should see the following message in the
+terminal:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS v1.14.0-rc1 *****
+   Hello World! hexiwear_kw40z
```
Comment (nordicjm) [inline boards/nxp/hexiwear/doc/index.rst:485]: Board name will not have soc in it anymore

File: boards/nxp/hexiwear/doc/index.rst [hunk 15]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+
+Debugging
+=========
+
+Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
```
Comment (nordicjm) [inline boards/nxp/hexiwear/doc/index.rst:494]: As above

File: boards/nxp/hexiwear/doc/index.rst [hunk 16]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+   :goals: debug
+
+Run telnet as shown earlier, step through the application in your debugger, and
+you should see the following message in the terminal:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS v1.14.0-rc1 *****
+   Hello World! hexiwear_kw40z
```
Comment (nordicjm) [inline boards/nxp/hexiwear/doc/index.rst:503]: As above

File: boards/nxp/rddrone_fmuk66/Kconfig.rddrone_fmuk66 [hunk 17]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_RDDRONE_FMUK66
-	bool "NXP RDDRONE-FMUK66"
-	depends on SOC_SERIES_KINETIS_K6X
-	select SOC_PART_NUMBER_MK66FN2M0VLQ18
+	select SOC_MK66F18
+
+config SOC_PART_NUMBER
```
Comment (nordicjm) [inline boards/nxp/rddrone_fmuk66/Kconfig.rddrone_fmuk66:7]: As above

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f [hunk 18]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019 Vestas Wind Systems A/S
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_TWR_KE18F
+	select SOC_MKE18F16
+
+config SOC_PART_NUMBER
+	default "MKE18F512VLL16"
```
Comment (nordicjm) [inline boards/nxp/twr_ke18f/Kconfig.twr_ke18f:8]: As above

File: boards/nxp/twr_kv58f220m/Kconfig.twr_kv58f220m [hunk 19]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019 SEAL AG
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_TWR_KV58F220M
+	select SOC_MKV58F24
+
+config SOC_PART_NUMBER
+	default "MKV58F1M0VLQ24"
```
Comment (nordicjm) [inline boards/nxp/twr_kv58f220m/Kconfig.twr_kv58f220m:8]: As above

File: boards/nxp/usb_kw24d512/Kconfig.usb_kw24d512 [hunk 20]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2017, Phytec Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_USB_KW24D512
+	select SOC_MKW24D5
+
+config SOC_PART_NUMBER
+	default "MKW24D512VHA5"
```
Comment (nordicjm) [inline boards/nxp/usb_kw24d512/Kconfig.usb_kw24d512:8]: As above

File: boards/segger/ip_k66f/Kconfig.ip_k66f [hunk 21]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_IP_K66F
-	bool "Segger IP-K66F"
-	depends on SOC_SERIES_KINETIS_K6X
-	select SOC_PART_NUMBER_MK66FN2M0VMD18
+	select SOC_MK66F18
+
+config SOC_PART_NUMBER
+	default "MK66FN2M0VMD18"
```
Comment (nordicjm) [inline boards/segger/ip_k66f/Kconfig.ip_k66f:8]: As above

File: soc/nxp/kinetis/Kconfig.soc [hunk 22]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2016 Intel Corporation
```
Comment (iuliana-prodan) [inline soc/nxp/kinetis/Kconfig.soc:1]: This seems like a new file. If this is the case the copyright should be updated.

File: soc/nxp/kinetis/k6x/Kconfig.defconfig [hunk 23]
```diff
@@ -1,5 +1,6 @@
-# Kinetis K6x series configuration options
+# Kinetis K6x series default configurations
 
+# Copyright 2024 NXP
 # Copyright (c) 2014-2016 Wind River Systems, Inc.
 # SPDX-License-Identifier: Apache-2.0
```
Comment (iuliana-prodan) [inline soc/nxp/kinetis/k6x/Kconfig.defconfig:5]: Based on [this](https://github.com/zephyrproject-rtos/zephyr/pull/68778/commits/6d7b5706751f7367c1b14eab82d2a442bffa5b1d#r1483946471) comment, the SOC_SERIES from this defconfig should be moved to Kconfig. soc

LE: the same for k8x, ke1xf, kv5x

File: boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z [hunk 24]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2017, NXP
```
Comment (iuliana-prodan) [inline boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z:1]: copyright update

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f [hunk 25]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019 Vestas Wind Systems A/S
+# SPDX-License-Identifier: Apache-2.0
```
Comment (iuliana-prodan) [inline boards/nxp/twr_ke18f/Kconfig.twr_ke18f:2]: copyright update?

**Comment by decsny** [comment]
@nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?

**Comment by nordicjm** [comment]
> @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?

The soc's Kconfig.soc file

**Comment by decsny** [comment]
> > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> 
> The soc's Kconfig.soc file

so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?

**Comment by nordicjm** [comment]
> > > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> > 
> > 
> > The soc's Kconfig.soc file
> 
> so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?

See https://github.com/zephyrproject-rtos/zephyr/pull/68559/files for an example

**Comment by decsny** [comment]
> > > > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> > > 
> > > 
> > > The soc's Kconfig.soc file
> > 
> > 
> > so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?
> 
> See https://github.com/zephyrproject-rtos/zephyr/pull/68559/files for an example

This just seems like an extraneous intermediate step, what would be the purpose?

**Comment by danieldegrasse** [comment]
> This just seems like an extraneous intermediate step, what would be the purpose? I suppose it is more clear what value the board should set?

I'd also like some clarification here. There are a few reasons we have the board selecting the SOC part number within the current port, versus the `config SOC_PART_NUMBER_xxx` scheme we use in v1: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_kinetis/k2x/Kconfig.soc#L51:

- The build error generated is the same (at least in my tests). Given this, I don't see how the `config SOC_PART_NUMBER_xxx` values at the SOC level will make it simpler for a customer to port a new board
- NXP tends to produce a lot of part numbers for a given SOC. See here for example (from [this file](https://raw.githubusercontent.com/zephyrproject-rtos/hal_nxp/master/mcux/mcux-sdk/devices/MIMXRT1062/MIMXRT1062.h)): 
```
Processors:          MIMXRT1062CVJ5A
**                          MIMXRT1062CVJ5B
**                          MIMXRT1062CVL5A
**                          MIMXRT1062CVL5B
**                          MIMXRT1062DVJ6A
**                          MIMXRT1062DVJ6B
**                          MIMXRT1062DVL6A
**                          MIMXRT1062DVL6B
**                          MIMXRT1062DVN6B
**                          MIMXRT1062XVN5B
```
My concerns with encoding the SOC part number as a Kconfig are the following:
  - typos are easy to make, and hard to catch. We can only really validate the Zephyr port on boards we have access to, which often means only one part number value really gets tested. A typo in tree will require the customer to send a patch or make a downstream fork of Zephyr to correct the issue.
  - Allowing customers to set the part number directly in the board does not expose the value as a configurable option within `menuconfig`, but it allows the customer to support whatever part number they have, regardless of if NXP has defined a `CONFIG_SOC_PART_NUMBER_xxx` value for it (which may take a bit, for example if a new silicon revision is released and we are slow to enable the EVK)

**Review by tejlmand — CHANGES_REQUESTED** [review]
@danieldegrasse @decsny @nordicjm thanks for raising an important discussion.

So far no specific decisions has been taken or raised in relation to SoC part number and hwmv2, so we have just continued with the design of v1. On my part, because I wasn't aware that someone had found existing solution too limited.

I will try to provide some input to allow us to move forward.

Before getting into details, then I want to highlight the importance of consistency, so that if we should change for nxp, then we should also change for other vendors.

> There are a few reasons we have the board selecting the SOC part number within the current port, versus the config SOC_PART_NUMBER_xxx scheme we use in v1

but in #68778 and HWMv1 the board is still selecting the SoC part number, but they just do it in two different ways.
In HWMv1 as `select SOC_PART_NUMBER_MK22FN512VLH12` which turns into:
```
SOC_PART_NUMBER_KINETIS_K2X
    string
    default "MK22FN512VLH12" if SOC_PART_NUMBER_MK22FN512VLH12
```
benefit of this, compared to #68778 is the ci check, as example in HWMv1, a typo like this `select SOC_PART_NUMBER_MK22FN512VLH21` (end digit 2 <--> 1 are swapped)

will result in error like:
```
warning: undefined symbol SOC_PART_NUMBER_MK22FN512VLH21:
- Referenced at boards/arm/frdm_k22f/Kconfig.board:6:
```

but with the proposal:
```
config SOC_PART_NUMBER
	default "MK22FN512VLH12"
```
no warning will ever be raised because `SOC_PART_NUMBER` is a string.
Thus, it very easy for a board maintainer to make an invalid string.

When the Kconfig SoC provides a Kconfig symbol and defines the string, then there is a lower risk of errors in the board Kconfig, as an error in the select will result in a warning.
Of course a typo in the Kconfig soc string is not caught directly, but upstream code goes through more reviews and more eyes are looking at the code, so chances are higher that mistakes are caught early.

Also, just looking upstream at a line like:
```
default "MK22FN512VLH21" if SOC_PART_NUMBER_MK22FN512VLH12
```
is faster to spot that something is mismatching, either the string or the setting.

Whereas
```
default "MK22FN512VLH21"
```
is not showing a double string, and no relation to what could be wrong.

> typos are easy to make, and hard to catch.

and that just got worse with this PR, se explanation above.

> A typo in tree will require the customer to send a patch or make a downstream fork of Zephyr to correct the issue.

And he should not send a patch if a Zephyr in-tree board accidentally writes: `default "MK22FN512VLH21"` ?
To me it seems like just pushing mistakes to other files, plus increasing the risk because all in-tree boards using same part number now defines the value of the string --> typos are not caught.
Whereas current scheme actually ensures a valid symbol is selected, and then that symbol sets the part number.
Meaning boards always selects a valid symbol (or CI will fail), thus an error in the string only has to be fixed in the SoC Kconfig.
An error in the soc setting name will need to be updated in all boards, but as those will now show up in CI as selecting an invalid soc part number, then those are very easy to find and fix.
An error in the string approach in this PR must be searched and replaced manually, and open PRs will go unnoticed, and as such PRs are not failing CI, then they'll risk not being fixed.

> Allowing customers to set the part number directly in the board does not expose the value as a configurable option within menuconfig, but it allows the customer to support whatever part number they have, regardless of if NXP has defined a CONFIG_SOC_PART_NUMBER_xxx value for it (which may take a bit, for example if a new silicon revision is released and we are slow to enable the EVK)

and this is still possible with existing method, both v1 and v2.

I assume we're talking oot boards, cause a new in-tree board should of course update the known SoC part numbers.

An oot board can still do:
```
# oot Kconfig.<board> file
config SOC_PART_NUMBER
        default "oot_defined_soc_part_number"
```

or even go for the more full fledged solution:
```
# oot Kconfig.<board> file
config SOC_PART_NUMBER_OOT
        bool

config SOC_PART_NUMBER
        default "oot_defined_soc_part_number" if SOC_PART_NUMBER_OOT

config BOARD_FOO
        select SOC_PART_NUMBER_OOT
```
making easy to remove custom oot soc part definition when an upstream symbol is made available.

Also having a `SOC_PART_NUMBER_x` symbol available makes it possible to test for the part number if need be, compared to string comparison.
And remember, `CONFIG_MISPELLED_SETTING` in code is also caught by ci, as:
```
Found references to undefined Kconfig symbols. If any of these are false
positives, then add them to UNDEF_KCONFIG_WHITELIST in ./scripts/ci/check_compliance.py.
...
CONFIG_SOC_PART_NUMBER_MK22FN512VLH21      <file>:<line>
```

based on above, I'm aligned with @nordicjm that current design should be kept, and I have seen no compelling reasons why the soc part number definition should change.

**Comment by danieldegrasse** [comment]
> benefit of this, compared to https://github.com/zephyrproject-rtos/zephyr/pull/68778 is the ci check, as example in HWMv1, a typo like this select SOC_PART_NUMBER_MK22FN512VLH21 (end digit 2 <--> 1 are swapped)

Sure, I was aware of the CI check. Given that this is a benefit for us (IE upstream developers), I'll update this PR to use `select SOC_PART_NUMBER`. However, I also expected that using `select SOC_PART_NUMBER_xxx` with an invalid Kconfig would give the user a useful error (IE by stopping the build at that point). At least in my tests, it does not appear to. Do you know why this is the case? 

When I build [this branch](https://github.com/nxp-zephyr/zephyr/tree/demo/k22f-part-num) for the `frdm_k22f`, I don't see any error related to the undefined `SOC_PART_NUMBER_xxx` Kconfig the board is selecting, but the build fails. If we set an invalid Kconfig in the board's `defconfig` file, then the build does give a warning. I guess my core question here is why does this difference exist, and can we make the build warn (or error out) if a board's Kconfig `select`s an invalid SOC part number?

**Comment by tejlmand** [comment]
> However, I also expected that using select SOC_PART_NUMBER_xxx with an invalid Kconfig would give the user a useful error (IE by stopping the build at that point). At least in my tests, it does not appear to. Do you know why this is the case?

Yes, that is because when compliance is run, then `KCONFIG_WARN_UNDEF=y` is set.
But that is default off when running a regular build.

One reason for this is because some Kconfig files / trees are conditionally sourced, such as the `arch` and `soc` trees.
Those trees were only sourced if the arch determined from board folder name matched the arch / soc tree in question.

This further means there could be `select`s in rest of Kconfig tree which would not be available in a particular build, but would be available in compliance where arch and SoC in HWMv1 was sourced using `*` as glob pattern.

Another possibility is use of `selects` into symbols of module's Kconfig files (not possible upstream) but where a module could be removed from manifest.

With the new approach in HWMv2, it could be valuable to see if we can improve the warnings in a normal build, so that undef warnings are also printed, but that is outside the scope of current porting work.

**Comment by danieldegrasse** [comment]
> This further means there could be selects in rest of Kconfig tree which would not be available in a particular build, but would be available in compliance where arch and SoC in HWMv1 was sourced using * as glob pattern.

Ok, this makes perfect sense- thanks for clarifying the reasoning. The PR should be updated to follow current guidance with `select SOC_PART_NUMBER_xxx`

**Comment by dleach02** [comment]
@nordicjm @tejlmand, are you good with the changes in response to your feedback? We want to close on this so we can start upstreaming additional soc groups.

@danieldegrasse please rebase your branch

File: soc/nxp/kinetis/ke1xf/Kconfig.soc [hunk 26]
```diff
@@ -1,28 +1,38 @@
...
-	bool "MKE14F16"
+	bool
+	select SOC_SERIES_KINETIS_KE1XF
 
 config SOC_MKE16F16
-	bool "MKE16F16"
+	bool
+	select SOC_SERIES_KINETIS_KE1XF
+	select HAS_MCUX
 	select HAS_MCUX_FLEXCAN
```
Comment (nordicjm) [inline soc/nxp/kinetis/ke1xf/Kconfig.soc:22]: these selections go in `Kconfig`

File: soc/nxp/kinetis/ke1xf/Kconfig.soc [hunk 27]
```diff
@@ -1,28 +1,38 @@
...
 
 config SOC_MKE18F16
-	bool "MKE18F16"
+	bool
+	select SOC_SERIES_KINETIS_KE1XF
+	select HAS_MCUX
 	select HAS_MCUX_FLEXCAN
```
Comment (nordicjm) [inline soc/nxp/kinetis/ke1xf/Kconfig.soc:28]: as above

File: soc/nxp/kinetis/k2x/Kconfig.soc [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+	default "mk22f12" if SOC_MK22F12
+
+config SOC_PART_NUMBER_MK22FN512VLH12
+	bool
+
+config SOC_PART_NUMBER_MK22FX512VLQ12
+	bool
+
+config SOC_PART_NUMBER
+	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/k2x/Kconfig.soc:39]:
```suggestion
```

File: soc/nxp/kinetis/k8x/Kconfig.soc [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+	bool
+
+config SOC_PART_NUMBER_MK82FN256VDC15
+	bool
+
+config SOC_PART_NUMBER_MK82FN256VLL15
+	bool
+
+config SOC_PART_NUMBER
+	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/k8x/Kconfig.soc:39]:
```suggestion
```

File: soc/nxp/kinetis/ke1xf/Kconfig.soc [hunk 30]
```diff
@@ -58,7 +68,7 @@ config SOC_PART_NUMBER_MKE18F512VLH16
 config SOC_PART_NUMBER_MKE18F512VLL16
 	bool
 
-config SOC_PART_NUMBER_KINETIS_KE1XF
+config SOC_PART_NUMBER
 	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/ke1xf/Kconfig.soc:72]:
```suggestion
```

File: soc/nxp/kinetis/kl2x/Kconfig.soc [hunk 31]
```diff
@@ -59,7 +53,7 @@ config SOC_PART_NUMBER_MKL25Z64VLK4
 config SOC_PART_NUMBER_MKL25Z128VLK4
 	bool
 
-config SOC_PART_NUMBER_KINETIS_KL2X
+config SOC_PART_NUMBER
 	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/kl2x/Kconfig.soc:57]:
```suggestion
```

File: soc/nxp/kinetis/kv5x/Kconfig.soc [hunk 32]
```diff
@@ -41,7 +47,7 @@ config SOC_PART_NUMBER_MKV58F1M0VLL24
 config SOC_PART_NUMBER_MKV58F1M0VLQ24
 	bool
 
-config SOC_PART_NUMBER_KINETIS_KV5X
+config SOC_PART_NUMBER
 	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/kv5x/Kconfig.soc:51]:
```suggestion
```

File: soc/nxp/kinetis/kwx/Kconfig.soc [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+	bool
+
+config SOC_PART_NUMBER_MKW41Z256VHT4
+	bool
+
+config SOC_PART_NUMBER_MKW41Z512VHT4
+	bool
+
+config SOC_PART_NUMBER
+	string
```
Comment (nordicjm) [inline soc/nxp/kinetis/kwx/Kconfig.soc:51]:
```suggestion
```

File: boards/nxp/twr_kv58f220m/doc/twr_kv58f220m.jpg
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply boards/nxp/twr_kv58f220m/doc/twr_kv58f220m.jpg:1]: Ok- left the image untouched

File: boards/nxp/hexiwear/doc/index.rst [hunk 34]
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
...
+| RTT       | on-chip    | console                             |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | soc flash                           |
++-----------+------------+-------------------------------------+
+| TRNG      | on-chip    | entropy                             |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/nxp/hexiwear/hexiwear_mkw40z4_defconfig``
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply boards/nxp/hexiwear/doc/index.rst]: Fixed

File: soc/nxp/kinetis/soc.yml [hunk 35]
```diff
@@ -0,0 +1,31 @@
+family:
+- name: nxp_kinetis
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply soc/nxp/kinetis/soc.yml]: Updated family name to `kinetis` in YAML

File: soc/nxp/kinetis/ke1xf/Kconfig.soc (hunk 26 above)
Comment (danieldegrasse) [reply soc/nxp/kinetis/ke1xf/Kconfig.soc]: Moved

**Review by manuargue — COMMENTED** [review]
shall we add `boards/nxp/index.rst` containing the toctree?

**Comment by dleach02** [comment]
@nordicjm @tejlmand Daniel has responded the the last round of comments. I don't want to keep pinging on this but given the stated objectives to have this branch ready for merge soon after v3.6 we are using the Kinetis PR as a pipe-cleaner before submitting our other platforms.

**Comment by dleach02** [comment]
> shall we add `boards/nxp/index.rst` containing the toctree?

done

**Comment by nordicjm** [comment]
Compliance issue needs fixing
Old board names in tests/samples need updating

**Review by tejlmand — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**